tools/testing/upload_artifacts.py

```diff
@@ -28,11 +30,25 @@ def zip_artifact(file_name: str, paths: list[str]) -> None:
                 f.write(file, os.path.relpath(file, REPO_ROOT))
 
 
-def upload_to_s3_artifacts() -> None:
+def concated_logs() -> str:
+    """Concatenate all the logs in the test-reports directory into a single string."""
+    logs = []
+    for log_file in glob.glob(f"{REPO_ROOT}/test/test-reports/**/*.log", recursive=True):
+        logs.append(f"=== {log_file} ===")
```

Comment:
What do you think about the possibility of not concatenating these logs but uploading as they are using something like `aws s3 rsync`? The log classifier then only need to handle new files.  This also open up the possibility of showing only the log where the failure comes from on HUD.